frontend/app/routes/protected/renew/$id/confirm-home-address.tsx

```diff
@@ -50,12 +50,20 @@ export async function loader({ context: { appContainer, session }, params, reque
   const countryList = appContainer.get(TYPES.domain.services.CountryService).listAndSortLocalizedCountries(locale);
   const regionList = appContainer.get(TYPES.domain.services.ProvinceTerritoryStateService).listAndSortLocalizedProvinceTerritoryStates(locale);
 
+  const homeAddressInfo = {
+    address: state.homeAddress?.address ?? state.clientApplication.contactInformation.homeAddress,
+    city: state.homeAddress?.city ?? state.clientApplication.contactInformation.homeCity,
+    province: state.homeAddress?.province ?? state.clientApplication.contactInformation.homeProvince,
+    postalCode: state.homeAddress?.postalCode ?? state.clientApplication.contactInformation.homePostalCode,
+    country: state.homeAddress?.country ?? state.clientApplication.contactInformation.homeCountry,
+    apartment: state.homeAddress?.apartment ?? state.clientApplication.contactInformation.homeApartment,
+  };
```

Comment:
This will cause similar issues to the partner information on the marital status page.

If a user has a postal code on file in `clientApplication` and switches to a country without a postal code, the `clientApplication`'s postal code will be used.

Instead, we can add a condition on the entire `state.homeAddress` object's presence:
```typescript
const homeAddressInfo = state.homeAddress ?
  {
    address: state.homeAddress.address,
    city: state.homeAddress.city,
    ...
  } : {
    address: state.clientApplication.contactInformation.homeAddress,
    city: state.clientApplication.contactInformation.homeCity,
    ...
  };

Reply:
(should be done for mailing address too)